### ISSUE TITLE
✅ 🐞 Element Caching prevents styles of Global widgets from loading - [ED-15101]

### ISSUE THREAD

**Issue by awebguy99** [elementor/elementor#27704]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

As per https://github.com/orgs/elementor/discussions/27371#discussioncomment-9789045

It looks like formatting in Global Widgets break when Element Caching enabled. I have a site in development (can't include the URL, confidential client info on it), all fine but when I enable Element Caching, any widget that is a Global Widget has its formatting break (wrong colour, wrong icon etc).

The widgets I've noticed so far breaking are Heading and Icon widgets.

1. If I disable Element Caching, the issue is fixed.
2. With Element Caching enabled, if I unlink the widget (I.E. change it from being a Global to normal widget), the issue is fixed.
3. I tried editing the Global Widget and disabling Element Caching but that does not fix the issue.
4. I also tried disabling all plugins, apart from Elementor and Ele Pro (all up to date). And I'm using Hello theme.

### Steps to reproduce

1. With Elements Caching Enabled, Create a page.
2. Add a Heading widget then add some styling (change the colour and size).
3. Publish and it will look fine on the front end.
4. Change the widget to a Global Widget, publish and now on the front end the colour and size will change.
5. Unlink the Global Widget / change it back to a normal widget, publish and it works again.

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.59 (Debian)
	MySQL version: mariadb.org binary distribution v10.5.25
	PHP Version: 8.1.28
	PHP Memory Limit: 512M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 450M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.5.4
	Site xxxxxxxxx
	Home xxxxxxxxx
	WP Multisite: No
	Max Upload Size: 450 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en_NZ
	Timezone: 12
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 3.0.2
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en-NZ
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.22.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.22.0
		Author: Elementor.com


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active
	Inline Font Icons: Inactive
	Additional Custom Breakpoints: Active by default
	Flexbox Container: Active
	Grid Container: Active
	Upgrade Swiper Library: Active
	Nested Elements Performance: Inactive by default
	Optimized Control Loading: Active
	Optimized Markup: Inactive by default
	Plugin Onboarding: Active by default
	Default to New Theme Builder: Active by default
	Hello Theme Header &amp; Footer: Inactive by default
	Elementor Home Screen: Active by default
	Editor Top Bar: Active
	Build with AI: Inactive
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Active
	Pages Panel: Inactive by default
	Element Caching: Active
	Conversion Center: Inactive by default
	Display Conditions: Active by default
	Form Submissions: Active by default
	Menu: Active
	Taxonomy Filter: Active by default


== Log ==
	
Log: showing 20 of 382024-05-21 17:04:43 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.21.2',
  'to' => '3.21.3',
)]
2024-05-21 17:04:43 [info] elementor::elementor_updater Started 
2024-05-21 17:04:43 [info] Elementor/Upgrades - _on_each_version Start  
2024-05-21 17:04:43 [info] Elementor/Upgrades - _on_each_version Finished 
2024-05-21 17:04:43 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.21.5',
  'to' => '3.21.6',
)]
2024-06-17 10:09:39 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.21.6',
  'to' => '3.22.0',
)]
2024-06-17 10:09:41 [info] elementor::elementor_updater Started 
2024-06-17 10:09:41 [info] Elementor/Upgrades - _on_each_version Start  
2024-06-17 10:09:41 [info] Elementor/Upgrades - _on_each_version Finished 
2024-06-17 10:09:41 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.21.6',
  'to' => '3.22.0',
)]
2024-06-17 11:44:40 [info] elementor-pro::elementor_pro_updater Started 
2024-06-17 11:44:40 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-06-17 11:44:40 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-06-17 11:44:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.21.3',
  'to' => '3.22.0',
)]
2024-06-17 11:44:40 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.21.3',
  'to' => '3.22.0',
)]
2024-06-18 10:11:09 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.22.0',
  'to' => '3.22.1',
)]
2024-06-18 10:21:23 [info] elementor::elementor_updater Started 
2024-06-18 10:21:23 [info] Elementor/Upgrades - _on_each_version Start  
2024-06-18 10:21:23 [info] Elementor/Upgrades - _on_each_version Finished 
2024-06-18 10:21:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.22.0',
  'to' => '3.22.1',
)]

JS: showing 2 of 2JS: 2024-05-01 01:55:15 [error X 3][../wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.21.4:2:56144] Model.isValidChild() should be implemented, please provide &#039;isValidChild&#039; functionality. 
JS: 2024-06-16 23:48:13 [error X 31][../wp-content/uploads/essential-addons-elementor/eael.js?ver=1714588957:28:489022] Cannot read properties of undefined (reading &#039;size&#039;) 

PHP: showing 2 of 2PHP: 2024-05-01 17:54:45 [warning X 2][../public_html/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::179] Undefined array key &quot;editor_post_id&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-06-17 11:48:07 [notice X 7][../wp-content/plugins/elementor/includes/base/controls-stack.php::229] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Incompatible

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by Optitesting** [comment]
I have the same issue, regarding global widgets like author, comments and table of contents. Settings is lost when activating element caching.

**Comment by steveindzine** [comment]
Same here. I am using a seperator with an icon in the middle of a short line, with element caching turned it just shows a small black square where the separator should be.

**Comment by Optitesting** [comment]
I have tried to deactivate caching at widget level but that could not fix it. Activating at widget level fixed some of the widget settings, but comments widget is not cured. Also when activating element caching function in comment widget, comments for random page is shown where comment widget is not used at all, like on start page.

**Comment by awebguy99** [comment]
Blows my mind this wasn't found in alpha stage. I mean, as an Elementor dev, surely you just need a page with all Pro widgets and all Pro features (like Global Widgets) on it. When developing a new feature, you simple make sure everything works before it even gets to Beta?

**Comment by Optitesting** [comment]
Agree on that. Can it be so that we are the test pilots, or have some plugins that E didnt test with? I have rather few plugins and not the most famous of those that make issues from time to time.

**Comment by nicholaszein** [comment]
Hello @awebguy99 (and others)! 

Thank you for submitting your issue! 🙏 

We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

Best regards

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.22.1! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
